_opensearch/search/paginate.md

```diff
@@ -0,0 +1,274 @@
+---
+layout: default
+title: Paginate results
+parent: Searching data
+nav_order: 21
+---
+
+## Paginate results
+
+You can use the following ways to paginate search results in OpenSearch: 
+
+1. The [`from` and `size` parameters](#the-from-and-size-parameters).
+1. [Scroll search](#scroll-search).
+1. The [`search_after` parameter](#the-search_after-parameter).
+
+## The `from` and `size` parameters
+
+The `from` and `size` parameters return results to your users one page at a time.
+
+The `from` parameter is the document number that you want to start showing the results from. The `size` parameter is the number of results that you want to show. Together, they let you return a subset of the search results.
+
+For example, if the value of `size` is 10 and the value of `from` is 0, you see the first 10 results. If you change the value of `from` to 10, you see the next 10 results (because the results are zero-indexed). So, if you want to see results starting from result 11, `from` must be 10.
+
+```json
+GET shakespeare/_search
+{
+  "from": 0,
+  "size": 10,
+  "query": {
+    "match": {
+      "play_name": "Hamlet"
+    }
+  }
+}
+```
+
+Use the following formula to calculate the `from` parameter relative to the page number:
+
+```json
+from = size * (page_number - 1)
+```
+
+Each time the user chooses the next page of the results, your application needs to make the same search query with an incremented `from` value.
+
+You can also specify the `from` and `size` parameters in the search URI:
+
+```json
+GET shakespeare/_search?from=0&size=10
+```
+
+If you only specify the `size` parameter, the `from` parameter defaults to 0.
+
+Querying for pages deep in your results can have a significant performance impact, so OpenSearch limits this approach to 10,000 results.
+
+The `from` and `size` parameters are stateless, so the results are based on the latest available data.
+This can cause inconsistent pagination.
+For example, assume a user stays on the first page of the results for a minute and then navigates to the second page; in that time, a new document is indexed in the background which is relevant enough to show up on the first page. In this scenario, the last result of the first page is pushed to the second page, so the user ends up seeing a result on the second page that they already saw on the first page.
+
+Use the `scroll` operation for consistent pagination. The `scroll` operation keeps a search context open for a certain period of time. Any data changes do not affect the results during this time.
+
+
+## Scroll search
+
+The `from` and `size` parameters allow you to paginate your search results, but with a limit of 10,000 results at a time.
+
+If you need to request massive volumes of data from, for example, a machine learning job, use the `scroll` operation instead. The `scroll` operation allows you to request an unlimited number of results.
```

Comment:
What is considered "massive volumes of data?" 1 million records? 1PB? I recommend clarifying.

Reply:
@Naarcha-AWS - do you have any suggestions on how to clarify here?

Reply:
I would say 1PB, because Scroll operates in a similar way to a data stream.